### PR TITLE
feat: capture speed up transaction

### PR DESCRIPTION
**Pull request by hanszhoou**
closes #

### CODE REVIEW

File: packages/web3-shared/src/hooks/useSpeedUpTransaction.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    state: TransactionState,
+    from: string,
+    contractData: Options | undefined,
+    transactionFunctionName: string,
+    timeout: number,
+) {
+    const web3 = useWeb3()
+    const [speedUpTx, setSpeedUpTx] = useState<Transaction | null>(null)
+    const [openTimer, setOpenTimer] = useState<NodeJS.Timeout | null>(null)
+    const interFace = contractData
```
Comment (guanbinrui) [inline packages/web3-shared/src/hooks/useSpeedUpTransaction.ts:19]:
```suggestion
    const interface = contractData
```

Comment (zhouhanseng) [reply packages/web3-shared/src/hooks/useSpeedUpTransaction.ts:19]: interface is a a reserved word.

**Comment by wenluomask** [comment]
I speeded up contract interaction in Polygon, the tx changed but Mask's Recent Transactions was still showing the old tx, which is pending status.
Screen Shot 2021-09-28 at 9 32 38 PM

**Comment by zhouhanseng** [comment]
@wenluomask Only Implement for red packet claim and creation, since I am not familiar with 'Mask's Recent Transactions logic' yet.

**Comment by zhouhanseng** [comment]
@wenluomask Ready for Testing

**Comment by wenluomask** [comment]
Transaction's tx changed for red packet creation, but tax didn't change for claim and unlock token. In addition, please add claiming status to claim button.
Screen Shot 2021-09-29 at 4 08 15 PM

**Comment by zhouhanseng** [comment]
There's claiming status.

**Pull request closed without merging**